lib/iris/fileformats/grib/_load_convert.py

```diff
@@ -172,24 +174,24 @@ def _masker(item):
             result = result.data
     else:
         result = ma.masked
-        if value != _MDI and factor != _MDI:
+        if value not in _MDIs and factor not in _MDIs:
             result = _unscale(value, factor)
     return result
 
 
 # Regulations 92.1.4 and 92.1.5.
-_MDI = 2 ** 32 - 1
+_MDIs = [2 ** 31 - 1, 2 ** 32 - 1]
```

Comment:
I'd be more inclined to codify the "is masked" operation as a function than have them as magic numbers in this way - it will help us avoid dtype checking all over the place.